test/e2e/allocate.go

```diff
@@ -0,0 +1,289 @@
+	})
+
+	It("allocate don't work when podgroup is pending", func() {
+		ctx := initTestContext(options{
+			nodesNumLimit: 2,
+			nodesResourceLimit: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2000m"),
+				corev1.ResourceMemory: resource.MustParse("2048Mi")},
+		})
+		defer cleanupTestContext(ctx)
+
+		pgName := "pending-pg"
+		pg := &schedulingv1beta1.PodGroup{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ctx.namespace,
+				Name:      pgName,
+			},
+			Spec: schedulingv1beta1.PodGroupSpec{
+				MinMember:    1,
+				MinResources: &thirtyCPU,
+			},
+			Status: schedulingv1beta1.PodGroupStatus{
+				Phase: schedulingv1beta1.PodGroupPending,
+			},
+		}
+
+		_, err := ctx.vcclient.SchedulingV1beta1().PodGroups(ctx.namespace).Create(context.TODO(), pg, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		slot1 := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("500m"),
+			corev1.ResourceMemory: resource.MustParse("512Mi")}
+
+		job := &jobSpec{
+			tasks: []taskSpec{
+				{
+					img: defaultNginxImage,
+					req: slot1,
+					min: 1,
+					rep: 1,
+				},
+			},
+		}
+
+		job.name = "exist"
+		existJob := createJob(ctx, job)
+		err = waitJobReady(ctx, existJob)
+		Expect(err).NotTo(HaveOccurred())
+
+		job.name = "allocate"
+		allocateJob := createJobWithCondition(ctx, job, pgName, nil)
+		err = waitJobPending(ctx, allocateJob)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("allocate don't work when queue NOT exist", func() {
+		q1 := "q10"
+		ctx := initTestContext(options{
+			nodesNumLimit: 2,
+			nodesResourceLimit: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2000m"),
+				corev1.ResourceMemory: resource.MustParse("2048Mi")},
+		})
+		defer cleanupTestContext(ctx)
+
+		slot1 := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("1000m"),
+			corev1.ResourceMemory: resource.MustParse("1024Mi")}
+
+		job := &jobSpec{
+			tasks: []taskSpec{
+				{
+					img: defaultNginxImage,
+					req: slot1,
+					min: 1,
+					rep: 1,
+				},
+			},
+		}
+
+		job.name = "j1-q1"
+		job.queue = q1
+		_, err := createJobInner(ctx, job)
+		Expect(err).To(HaveOccurred(), "queue not found")
+	})
+
+	It("allocate don't work when job NOT satisify predicates", func() {
+		ctx := initTestContext(options{
+			nodesNumLimit: 2,
+			nodesResourceLimit: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2000m"),
+				corev1.ResourceMemory: resource.MustParse("2048Mi")},
+		})
+		defer cleanupTestContext(ctx)
+
+		slot1 := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("1000m"),
+			corev1.ResourceMemory: resource.MustParse("1024Mi")}
+
+		job := &jobSpec{
+			tasks: []taskSpec{
+				{
+					img: defaultNginxImage,
+					req: slot1,
+					min: 1,
+					rep: 1,
+				},
+			},
+		}
+
+		job.name = "j1-n3"
+		node3Job := createJobWithCondition(ctx, job, "", map[string]string{
+			corev1.LabelHostname: "node3",
```

Comment:
how do you get "node3"

Reply:
Do you mean a non-exist node?

Reply:
I will use other conditions to test allocate when job not satisify predicates.